src/buildHTML.js

```diff
@@ -1473,6 +1501,230 @@ groupTypes.accent = function(group, options) {
     }
 };
 
+groupTypes.horizBrace = function(group, options) {
+    const style = options.style;
+    const ruleWidth = fontMetrics.metrics.defaultRuleThickness /
+        style.sizeMultiplier;
+
+    const haveNote = (group.type === "supsub");
+    let note;
+    let newOptions;
+    let noteReset;
+    if (haveNote) {
+        if (group.value.sup) {
+            newOptions = options.withStyle(style.sup());
+            note = buildGroup(group.value.sup, newOptions);
+            noteReset =  makeSpan([style.reset(), style.sup().cls()],
+                [note], newOptions);
+        } else {
+            newOptions = options.withStyle(style.sub());
+            note = buildGroup(group.value.sub, newOptions);
+            noteReset =  makeSpan([style.reset(), style.sub().cls()],
+                [note], newOptions);
+        }
+        group = group.value.base;
+    }
+
+    // Build the base group
+    const body = buildGroup(
+       group.value.base, options.withStyle(style.cramp()));
+
+    // If a custom bracket, convert units.
+    if (group.value.thickness) {
+        group.value.thickness.value = calculateSize(group.value.thickness.value,
+                options.style);
+    }
+    if (group.value.height) {
+        group.value.height = calculateSize(group.value.height.value,
+                options.style);
+    }
+
+    // Create the stretchy element
+    const braceBody = stretchy.svgSpan(group, options);
+
+    // Generate the vlist, with the appropriate kerns
+    const braceShift = (group.value.isOver ? -body.height - 3 * ruleWidth :
+                        body.depth + braceBody.height + 3 * ruleWidth);
+
+    let vlist = buildCommon.makeVList([
+        {type: "elem", elem: body, shift: 0},
+        {type: "elem", elem: braceBody, shift: braceShift},
+    ], "individualShift", null, options);
```

Comment:
No particular reason for `shift`. For the extensible arrows, I had to use `makeVList` in `individualShift` mode and I got accustomed to it. As far as I know, the behavior is the same. But your point is well taken. For braces, it would be more clear if I used `makeVList` in `firstBaseLine` mode. So I'll change it.

Well, I could move that block of course, but that would mean writing it twice.  The vlist in question consists of the subject matter and the brace: ![](http://quicklatex.com/cache3/b9/ql_f9cad1ba16020c3b7a5726ab45c349b9_l3.svg)

If there is a note needed above the brace, then I need to wrap the first vlist with another one to show the note: ![](http://quicklatex.com/cache3/2d/ql_cddfd68906a8c0bc88f1a16c94d06a2d_l3.svg)

And I can't put all three items in the same vlist. If I did, then a long note might control width of the brace. I want the brace width to be controlled only by the subject matter: ![](http://quicklatex.com/cache3/46/ql_fc7e24f7833a3275d064da0c759fc246_l3.svg)